src/Core/Domain/Product/ValueObject/ResponseCode.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Response code for product unavailable case.
+ */
+class ResponseCode
```

Comment:
Not sure about this VO, response code is very technical. :/

Reply:
but it does exist in Domain - even visible in the form. If it only would hold id then I would agree with you but this VO also validates available types